lib/rules/padding-between-test-functions.js

```diff
@@ -0,0 +1,142 @@
+
+'use strict';
+
+var names = [
+    'describe',
+    'context',
+    'it',
+    'specify',
+    'before',
+    'after',
+    'beforeEach',
+    'afterEach',
+    'suite',
+    'test',
+    'suiteSetup',
+    'suiteTeardown',
+    'setup',
+    'teardown'
+];
+
+/**
+ * Checks if a node is a test function expression, or object (e.g., it() or it.only())
+ * @param {Node} node node to check
+ * @returns {boolean} True if the node is a test function expression
+ */
+function isTestFunction(node) {
+  return (
+    node &&
+    node.expression &&
+    node.expression.callee && (
+        names.indexOf(node.expression.callee.name) > -1 ||
+        node.expression.callee.object && names.indexOf(node.expression.callee.object.name) > -1
+    )
+  );
+}
+
+/**
+ * Checks if there is padding between two tokens
+ * @param {Token} first The first token
+ * @param {Token} second The second token
+ * @returns {boolean} True if there is at least a line between the tokens
+ */
+function isPaddingBetweenTokens(first, second) {
+    return second.loc.start.line - first.loc.end.line >= 2;
+}
+
+module.exports = {
+    meta: {
+        docs: {
+            description: 'Enforce spacing between test functions',
+            category: 'Stylistic Issues',
+            recommended: false
+        },
+        fixable: 'whitespace',
+        schema: [ {
+            oneOf: [ {
+                enum: [ 'always', 'never' ]
+            } ]
+        } ]
+    },
+    create: function (context) {
+        var opt = context.options[0] || 'always',
+        message = opt === 'never'
+            ? 'Test functions must not be padded by blank lines.'
+            : 'Test functions must be padded by blank lines.',
+        sourceCode = context.getSourceCode();
+
+        return {
+            ExpressionStatement: function (node) {
+                var firstToken = sourceCode.getFirstToken(node),
+                    tokenBeforeFirst = sourceCode.getTokenBefore(firstToken, { includeComments: true }),
+                    lastToken = sourceCode.getLastToken(node),
+                    tokenAfterLast = sourceCode.getTokenAfter(lastToken, { includeComments: true });
+
+                function shouldReturn() {
+                    return (
+                        !sourceCode.getTokenBefore(node) ||
+                        !sourceCode.getTokenAfter(node) ||
```

Comment:
This may be indicative of the complexity of this function - but it'll check both before and after a given mocha function.

So in this case, it checks above describe finds no tokens & aborts; checks below, finds `somethingElse` without padding and reports (if options match)

I've split this out a bit into `topChecker` and `bottomChecker` -- this helps the complexity of the function and hopefully makes things a bit clearer.

I've added test cases to check this particular case.